be/src/formats/orc/orc_mapping.cpp

```diff
@@ -0,0 +1,149 @@
+            std::string field_name = format_column_name(orc_type.getFieldName(i), case_sensitive);
+            tmp_orc_fieldname_2_pos[field_name] = i;
+        }
+
+        for (size_t index = 0; index < origin_type.children.size(); index++) {
+            std::string field_name = format_column_name(origin_type.field_names[index], case_sensitive);
+            auto it = tmp_orc_fieldname_2_pos.find(field_name);
+            if (it == tmp_orc_fieldname_2_pos.end()) {
+                auto s = strings::Substitute(
+                        "OrcChunkReader::_set_child_mapping "
+                        "not found struct subfield $0",
+                        field_name);
+                return Status::NotFound(s);
+            }
+            const orc::Type& orc_child_type = *orc_type.getSubtype(it->second);
+            mapping->add_mapping(index, orc_child_type.getColumnId());
+
+            const TypeDescriptor& origin_child_type = origin_type.children[index];
+            if (origin_child_type.is_complex_type()) {
+                OrcMappingPtr child_mapping = std::make_shared<OrcMapping>();
+                mapping->add_child(index, child_mapping);
+                RETURN_IF_ERROR(_set_child_mapping(child_mapping, origin_child_type, orc_child_type, case_sensitive));
+            }
+        }
+    } else if (origin_type.type == PrimitiveType::TYPE_ARRAY) {
+        DCHECK(orc_type.getKind() == orc::TypeKind::LIST);
+        const TypeDescriptor& origin_child_type = origin_type.children[0];
+        const orc::Type& orc_child_type = *orc_type.getSubtype(0);
+
+        mapping->add_mapping(0, orc_child_type.getColumnId());
+
+        if (origin_child_type.is_complex_type()) {
+            OrcMappingPtr child_mapping = std::make_shared<OrcMapping>();
+            mapping->add_child(0, child_mapping);
+            RETURN_IF_ERROR(_set_child_mapping(child_mapping, origin_child_type, orc_child_type, case_sensitive));
+        }
+    } else if (origin_type.type == PrimitiveType::TYPE_MAP) {
+        DCHECK(orc_type.getKind() == orc::TypeKind::MAP);
+
+        mapping->add_mapping(0, orc_type.getSubtype(0)->getColumnId());
+        mapping->add_mapping(1, orc_type.getSubtype(1)->getColumnId());
+
+        // Map's key must be primitive type, so we only consider value.
+        const TypeDescriptor& origin_child_type = origin_type.children[1];
+
+        if (origin_child_type.is_complex_type()) {
+            OrcMappingPtr child_mapping = std::make_shared<OrcMapping>();
+            mapping->add_child(1, child_mapping);
+            const orc::Type& orc_child_type = *orc_type.getSubtype(1);
+            RETURN_IF_ERROR(_set_child_mapping(child_mapping, origin_child_type, orc_child_type, case_sensitive));
+        }
+    }
+    return Status::OK();
+}
+
+std::string OrcMapping::format_column_name(const std::string& col_name, bool case_sensitive) {
+    return case_sensitive ? col_name : boost::algorithm::to_lower_copy(col_name);
```

Comment:
prefer use in-house function instead of boost, such as `StringCaseEqual` in `be/src/util/string_util.h`

Reply:
That's used to compare two strings with case insensitive, `format_column_name()` is used to convert string based on case_sensitive.